multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/jobs/FinishedFlowableHistoricProcessesCleaner.java

```diff
@@ -0,0 +1,124 @@
+
+@Named
+public class FinishedFlowableHistoricProcessesCleaner {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FinishedFlowableHistoricProcessesCleaner.class);
+    private static final Marker LOG_MARKER = MarkerFactory.getMarker("finished-flowable-historic-processes-cleaner");
+    private static final int SELECTED_INSTANCE_FOR_CLEAN_UP = 1;
+    private static final int MAX_OPERATIONS_PER_PAGE = 100;
+
+    private final ApplicationConfiguration applicationConfiguration;
+    private final OperationService operationService;
+    private final HistoryService historyService;
+
+    @Inject
+    public FinishedFlowableHistoricProcessesCleaner(ApplicationConfiguration applicationConfiguration, OperationService operationService,
+                                                    HistoryService historyService) {
+        this.applicationConfiguration = applicationConfiguration;
+        this.operationService = operationService;
+        this.historyService = historyService;
+    }
+
+    @Scheduled(fixedRateString = "#{@applicationConfiguration.getExecutionTimeForFinishedProcesses()}")
+    public void cleanUp() throws ExecutionException, InterruptedException {
+        if (applicationConfiguration.getApplicationInstanceIndex() != SELECTED_INSTANCE_FOR_CLEAN_UP) {
+            return;
+        }
+        LOGGER.info(LOG_MARKER, Messages.CLEAN_UP_JOB_FOR_FINISHED_OPERATIONS_HAS_STARTED);
+        deleteFinishedProcesses();
+        LOGGER.info(LOG_MARKER, Messages.CLEAN_UP_JOB_FOR_FINISHED_OPERATIONS_HAS_FINISHED);
+    }
+
+    private void deleteFinishedProcesses() throws InterruptedException, ExecutionException {
+        Date endedBefore = new DateTime().minusHours(2)
+                                         .minusMinutes(30)
+                                         .toDate();
+        Date endedAfter = new DateTime().minusHours(5)
+                                        .toDate();
+        int page = 0;
+        int deletedProcesses = 0;
+        List<Operation> operations = getFinishedOperationsByPage(page, endedBefore, endedAfter);
+        ExecutorService executorService = Executors.newFixedThreadPool(3);
+        while (!operations.isEmpty()) {
+            List<Callable<Boolean>> historicProcessesDeletionTasks = getHistoricProcessInstances(operations).stream()
+                                                                                                            .map(historicProcessInstance -> (Callable<Boolean>) () -> deleteHistoricProcess(historicProcessInstance))
+                                                                                                            .collect(Collectors.toList());
+            List<Future<Boolean>> executedDeletionTasks = executorService.invokeAll(historicProcessesDeletionTasks);
+            for (Future<Boolean> executedDeletionTask : executedDeletionTasks) {
+                if (executedDeletionTask.get()) {
```

Comment:
Would it benefit to have a timeout for each job? 
This `get()` method will block until the job has finished.

Reply:
I do not think so, the same logic is used in the other flowable cleaner and seems stable.
The blocking is actually during "executorService.invokeAll".